Library/Homebrew/tap.rb

```diff
@@ -280,4 +317,75 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
+
+  undef_method :install
+  undef_method :uninstall
+  undef_method :command_files
+  undef_method :pinned_symlink_path
+  undef_method :pinned?
+  undef_method :pin
+  undef_method :unpin
+  undef_method :to_hash
```

Comment:
That's not a good idea; fundamentally this shows more that this is not a "real" Tap. Use a module and composition if you want to share functionality this way rather than inheritance where it does make sense.